pkg/services/librarypanels/librarypanels_test.go

```diff
@@ -178,12 +178,240 @@ func TestGetAllLibraryPanels(t *testing.T) {
 		})
 }
 
+func TestUpdateLibraryPanel(t *testing.T) {
+	testScenario(t, "When an admin tries to update a library panel that does not exist, it should fail",
+		func(t *testing.T, sc scenarioContext) {
+			cmd := updateLibraryPanelCommand{}
+			response := sc.service.updateHandler(sc.reqContext, cmd)
+			require.Equal(t, 404, response.Status())
+		})
+
+	testScenario(t, "When an admin tries to update a library panel that exists, it should succeed",
+		func(t *testing.T, sc scenarioContext) {
+			command := getCreateCommand(1, "Text - Library Panel")
+			response := sc.service.createHandler(sc.reqContext, command)
+			require.Equal(t, 200, response.Status())
+
+			var result libraryPanelResult
+			err := json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+
+			cmd := updateLibraryPanelCommand{
+				FolderID: 2,
+				Name:     "Panel - New name",
+				Model: []byte(`
+								{
+								  "datasource": "${DS_GDEV-TESTDATA}",
+								  "id": 1,
+								  "name": "Model - New name",
+								  "type": "text"
+								}
+							`),
+			}
+			sc.reqContext.ReplaceAllParams(map[string]string{":uid": result.Result.UID})
+			response = sc.service.updateHandler(sc.reqContext, cmd)
+			require.Equal(t, 200, response.Status())
+			err = json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+			require.Equal(t, int64(2), result.Result.FolderID)
+			require.Equal(t, "Panel - New name", result.Result.Name)
+			require.Equal(t, "Model - New name", result.Result.Model["name"])
```

Comment:
Just a suggestion, but I think it's better to use [go-cmp](https://github.com/google/go-cmp) here, as we do elsewhere in the project, to verify the whole object, not just selected fields.

Reply:
Can you exclude fields to be compared in go-comp?

Reply:
Yes, you can.

Reply:
That's splendid news! Thank you.

Reply:
Fixed